src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -83,33 +87,27 @@ private enum Flags : byte
             }
 
             _containingType = containingType;
-            _syntaxRef = syntax.GetReference();
-            _refKind = syntax.Type.GetRefKind();
+            _refKind = GetTypeSyntax(syntax).GetRefKind();
 
-            SyntaxTokenList modifiers = syntax.Modifiers;
+            SyntaxTokenList modifiers = GetModifierTokens(syntax);
             bodyBinder = bodyBinder.WithUnsafeRegionIfNecessary(modifiers);
             bodyBinder = bodyBinder.WithAdditionalFlagsAndContainingMemberOrLambda(BinderFlags.SuppressConstraintChecks, this);
 
-            var propertySyntax = syntax as PropertyDeclarationSyntax;
-            var arrowExpression = propertySyntax != null
-                ? propertySyntax.ExpressionBody
-                : ((IndexerDeclarationSyntax)syntax).ExpressionBody;
+            var arrowExpression = GetArrowExpression(syntax);
             bool hasExpressionBody = arrowExpression != null;
-            bool hasInitializer = !isIndexer && propertySyntax.Initializer != null;
-
-            GetAcessorDeclarations(syntax, diagnostics, out bool isAutoProperty, out bool hasAccessorList,
-                                   out AccessorDeclarationSyntax getSyntax, out AccessorDeclarationSyntax setSyntax);
-
-            bool accessorsHaveImplementation;
-            if (hasAccessorList)
-            {
-                accessorsHaveImplementation = (getSyntax != null && (getSyntax.Body != null || getSyntax.ExpressionBody != null)) ||
-                                              (setSyntax != null && (setSyntax.Body != null || setSyntax.ExpressionBody != null));
-            }
-            else
-            {
-                accessorsHaveImplementation = hasExpressionBody;
-            }
+            bool hasInitializer = HasInitializer(syntax);
+
+            GetAccessorDeclarations(
+                syntax,
+                diagnostics,
+                out bool isAutoProperty,
+                out bool hasAccessorList,
+                out bool accessorsHaveImplementation,
+                out bool hasGetAccessor,
+                out bool hasSetAccessor,
```

Comment:
Do these values need to be calculated by `GetAccessorDeclarations()`? It looks like this method assumes `hasGetAccessor == (getSyntax != null)`, and `hasSetAccessor == (setSyntax != null)`.

Consider removing those parameters and just checking `getSyntax` and `setSyntax` directly as before.

Reply:
That would be incorrect for SynthesizedRecordPropertySymbol

Reply:
`SynthesizedRecordPropertySymbol` sets the fields with:
```
getSyntax = setSyntax = syntax;
hasGetAccessor = hasSetAccessor = true;
```

---
In reply to: [445081736](https://github.com/dotnet/roslyn/pull/45411#discussion_r445081736) [](ancestors = 445081736)

Reply:
Yeah, you're right. I'll see if that would work.